src/client/components/Sidebar.tsx

```diff
@@ -53,11 +70,23 @@ class Sidebar extends PureComponent<SidebarProps, SidebarState> {
       <MainButton
         label="Search"
         onClick={async () => {
-          const { start, end } = this.state
+          const { start, end, timezone } = this.state
+          // Find offset between local timezone and UTC in milliseconds
+          const localTimezoneOffset = new Date().getTimezoneOffset() * 60 * 1000
+          // Find offset between selected timezone and UTC in milliseconds
+          const selectedTimezoneOffset = timezone.offset * 60 * 60 * 1000
+          // Adjust 'start' date from local timezone to selected timezone
+          const adjustedStart = new Date(
+            start.getTime() - localTimezoneOffset - selectedTimezoneOffset
+          )
+          // Adjust 'end' date from local timezone to selected timezone
+          const adjustedEnd = new Date(
+            end.getTime() - localTimezoneOffset - selectedTimezoneOffset
+          )
```

Comment:
This part should be a utility function with tests.
This repo doesn't have tests yet.
You can copy the setup from the info server.